apps/web/src/pages/brand/tabs/LayoutsListPage.tsx

```diff
@@ -37,16 +42,25 @@ export function LayoutsListPage() {
     setToDelete(id);
   };
 
-  const confirmDelete = () => {
-    handleDeleteLayout(toDelete);
+  const confirmDelete = async () => {
+    await handleDeleteLayout(toDelete);
     setToDelete('');
   };
 
-  const handleDeleteLayout = (layoutId: string) => {};
+  const handleDeleteLayout = async (layoutId: string) => {
+    try {
+      await deleteLayout(layoutId);
+      await refetchLayouts();
+      successMessage('Layout deleted!');
+    } catch (e: any) {
+      errorMessage(e?.message || 'Could not delete');
+    }
+  };
 
-  const goBack = () => {
-    setActiveScreen(ActivePageEnum.LAYOUTS_LIST);
+  const goBack = async () => {
     setEditId('');
+    setActiveScreen(ActivePageEnum.LAYOUTS_LIST);
+    await refetchLayouts();
```

Comment:
It surprises me to refetch the layouts every time that go back is executed. I find it a bit weird and out of place. I think that if we want to keep the list up to date after doing any operation there we should refetch the layouts list on the load of the `LayoutsListPage`. Therefore I would suggest to go for an `useEffect` implementation.